e2e_tests/tests/per_provider/normal_tests/create_destroy_key.rs

```diff
@@ -2,28 +2,37 @@
 // SPDX-License-Identifier: Apache-2.0
 use e2e_tests::TestClient;
 use parsec_client::core::interface::requests::ResponseStatus;
+use parsec_client::core::interface::requests::Result;
+#[cfg(not(feature = "cryptoauthlib-provider"))]
 use picky_asn1_x509::RSAPublicKey;
 
 #[test]
-fn create_and_destroy() {
+fn create_and_destroy() -> Result<()> {
     let mut client = TestClient::new();
     client.do_not_destroy_keys();
     let key_name = String::from("create_and_destroy");
 
-    client.generate_rsa_sign_key(key_name.clone()).unwrap();
-    client.destroy_key(key_name).unwrap();
+    #[cfg(not(feature = "cryptoauthlib-provider"))]
+    client.generate_rsa_sign_key(key_name.clone())?;
+    #[cfg(feature = "cryptoauthlib-provider")]
+    client.generate_ecc_key_pair_secpr1_ecdsa_sha256(key_name.clone())?;
+    client.destroy_key(key_name)
 }
 
+#[cfg(not(feature = "cryptoauthlib-provider"))]
+// This test nakes no sense for calib - it does not destroy created keys
```

Comment:
When generating a key, you can check if the mapping exist or return an error otherwise? Using the `does_not_exist` method.

Reply:
Legacy comment removed.